lib/natalie/compiler/transformers/match_required_node.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Natalie
+  class Compiler
+    module Transformers
+      class MatchRequiredNode
+        attr_reader :compiler, :value
+
+        def initialize(compiler)
+          @compiler = compiler
+        end
+
+        def method_missing(_method, node)
```

Comment:
I suspect you did this to find out what method Prism was calling... 😄

Reply:
No, that one is actually easier to do without a `method_missing`, since it will show the `NoMethodError` with all the information

I've been using the visitor pattern of Prism, every node type has an `accept` method that pretty much looks like this:
```ruby
def accept(visitor)
  visitor.send("visit_#{type}", self)
end
```
We always start with calling `visit_match_required_node`, where we use this pattern to visit the `accept` (RHS) part of the statement. We've got roughly three options here:
* ArrayPatternNode, for `expr => [1, y]`, this is not supported and raises an explicit exception
* LocalVariableTargetNode, for `expr => a`, this is handled as being an assignment with the visitor pattern in this class
* Anything else, which can be a class, an integer literal, a range, or pretty much anything: this is all rewritten from `expr => target` to something like `raise NoMatchBlah unless target === expr`, and the compiler ivar (the `Pass1` instance) is used to convert the `expr` and `target` to the relevant instructions (the calls to `compiler.transform_expression`)

The first commit has an explicit `alias visit_constant_read_node visit_constant_path_node` to handle both constants and constant paths, even though the method `visit_constant_path_node` has not a single reference to constants, everything is delegated to `Pass1` via `compiler.transform_expression`
The second commit introduces the `method_missing` handler, which handles the visitation of every not explicitly handled node type, which means we use this code in the general case and no longer have to worry about the node type.

it could probably be made a bit more secure, with `return super(method, node) if method != :"visit_#{node.type}"` and handling arities other than 1, but given that this code is close to being rewritten anyway, I think we're good for now.